pkg/apis/flowcontrol/validation/validation.go

```diff
@@ -73,25 +77,47 @@ var supportedLimitResponseType = sets.NewString(
 // ValidateFlowSchema validates the content of flow-schema
 func ValidateFlowSchema(fs *flowcontrol.FlowSchema) field.ErrorList {
 	allErrs := apivalidation.ValidateObjectMeta(&fs.ObjectMeta, false, ValidateFlowSchemaName, field.NewPath("metadata"))
-	allErrs = append(allErrs, ValidateFlowSchemaSpec(&fs.Spec, field.NewPath("spec"))...)
+	specPath := field.NewPath("spec")
+	switch fs.Name {
+	case flowcontrol.FlowSchemaNameExempt, flowcontrol.FlowSchemaNameCatchAll:
+		src := bootstrap.MandatoryFlowSchemaExempt
+		if fs.Name == flowcontrol.FlowSchemaNameCatchAll {
+			src = bootstrap.MandatoryFlowSchemaCatchAll
+		}
+		var rightSpec flowcontrol.FlowSchemaSpec
+		err := install.GetTheScheme().Converter().Convert(&src.Spec, &rightSpec, 0, &conversion.Meta{})
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(specPath, fs.Spec, "Conversion failed: "+err.Error()))
+		} else if !apiequality.Semantic.DeepEqual(fs.Spec, rightSpec) {
```

Comment:
Usually I'd complain about this, in that it's almost always overly restrictive. But I think it's right in this case to use deep equal. I think a comment that this is an exception to the rule might be good.

Reply:
done